app/services/search/query_builders/classified_listing.rb

```diff
@@ -0,0 +1,97 @@
+module Search
+  module QueryBuilders
+    class ClassifiedListing
+      TERM_KEYS = %i[
+        category
+        contact_via_connect
+        location
+        slug
+        tags
+        title
+        user_id
+      ].freeze
+
+      RANGE_KEYS = %i[
+        bumped_at
+        expires_at
+      ].freeze
+
+      QUERY_KEYS = %i[
+        classified_listing_search
+      ].freeze
+
+      DEFAULT_PARAMS = {
+        sort_by: "bumped_at",
+        sort_direction: "desc",
+        size: 0
+      }.freeze
+
+      attr_accessor :params, :body
+
+      def initialize(params)
+        @params = params.deep_symbolize_keys
+        build_body
+      end
+
+      def as_hash
+        @body
+      end
+
+      private
+
+      def build_body
+        @body = ActiveSupport::HashWithIndifferentAccess.new
+        build_queries
+        add_sort
+        # By default we will return 0 documents if size is not specified
+        @body[:size] = @params[:size] || DEFAULT_PARAMS[:size]
+      end
+
+      def build_queries
+        @body[:query] = {}
+        @body[:query][:bool] = { filter: filter_conditions }
+        @body[:query][:bool] = { must: query_conditions } if query_keys_present?
+      end
+
+      def add_sort
+        sort_key = @params[:sort_by] || DEFAULT_PARAMS[:sort_by]
+        sort_direction = @params[:sort_direction] || DEFAULT_PARAMS[:sort_direction]
+        @body[:sort] = {
+          sort_key => sort_direction
+        }
+      end
+
+      def filter_conditions
+        term_keys + range_keys
+      end
+
+      def term_keys
+        TERM_KEYS.map do |term_key|
+          next if @params[term_key].blank? && @params[term_key] != false
+
+          { term: { term_key => @params[term_key] } }
+        end.compact
+      end
+
+      def range_keys
+        RANGE_KEYS.map do |range_key|
+          next if @params[range_key].blank? && @params[range_key] != false
+
+          { range: { range_key => @params[range_key] } }
+        end.compact
+      end
+
+      def query_keys_present?
+        QUERY_KEYS.detect { |key| @params[key].present? }
+      end
+
+      def query_conditions
+        QUERY_KEYS.map do |query_key|
+          next if @params[query_key].blank?
+
+          { match: { query_key => { query: @params[query_key] } } }
```

Comment:
This might actually end up changing, I am in the middle of figuring out the exact type of query we want here to get the fuzzy search results we want. 

Here is what has been changing as I implement this with my frontend if you are curious where this is headed, https://github.com/thepracticaldev/dev.to/compare/mstruve/implement-chat-search#diff-c8d5692235cc1a81cb7ed64d40688e2c

Reply:
Ah, I see. Perhaps I jumped the gun with this PR then. How about I go ahead and pause this PR (close it out) until yours is opened and merged? That way I’ll have the latest strategy you’d like to use. It’s starting to feel like I’m chasing your work a little too closely, which then messes me up. I don’t mind holding off.

Reply:
Whatever works best for you! I'm game if you want to hold off and maybe tackle some refactoring of the index worker or other monitoring tickets.